File: src/lib/types/book.ts

export interface Book {
    id: string;
    userId: string;
    title: string;
    author: string;
    genre: string;
    condition: 'New' | 'Like New' | 'Very Good' | 'Good' | 'Fair' | 'Poor';
    available: boolean;
    description?: string;
    image?: string;
    createdAt: Date;
    updatedAt: Date;
}

export type BookFormData = Omit<Book, 'id' | 'userId' | 'createdAt' | 'updatedAt'>;

File: backend/models/book.py

from pydantic import BaseModel
from typing import Optional

class BookBase(BaseModel):
    title: str
    author: str
    genre: str
    condition: Optional[str] = None
    isAvailable: bool = True
    description: Optional[str] = None
    userId: str
    image_url: str

class BookCreate(BookBase):
    pass

class Book(BookBase):
    id: str

    class Config:
        from_attributes = True